extensions/interactions/AlgebraicExpressionInput/directives/algebraic-expression-input-rules.service.spec.ts

```diff
@@ -160,6 +160,110 @@ describe('Algebraic expression input rules service', () => {
       '-w+5z', {x: inputString})).toBeFalse();
   });
 
+  it('should have a correct MatchesUpToTrivialManipulations rule', () => {
+    inputString = '((x)^(2)-x)/(z)-4*y';
+
+    expect(algebraicRulesService.MatchesUpToTrivialManipulations(
+      '((x)^(2)-x)/(z)-4*y', {x: inputString})).toBeTrue();
+    expect(algebraicRulesService.MatchesUpToTrivialManipulations(
+      '((x)^(2)-x)/(z)-(8*y)/(2)', {x: inputString})).toBeTrue();
```

Comment:
Is this actually correct? /cc @BenHenning 

I thought "trivial manipulations" was meant to handle commutativity/associativity but not actual numeric evaluation. But lmk if I've gotten this wrong.

@BenHenning -- you might want to double-check and certify the other test cases in this file too.

Reply:
Also @BenHenning I can't find this case in your test sheet. Maybe worth adding it (and others in this file)?

Reply:
@seanlip The `MatchesUpToTrivialManipulations` rule is the same as the initially implemented `MatchesExactlyWith` rule. And as per that rule's definition, we allow numeric simplification for algebraic expressions and equations.
If we want to restrict that, we won't be able to use nerdamer since it automatically simplifies numeric parts.

Reply:
Hm -- just to check, then, would you then consider 2 * 2 * 2 * 3 to be the same as 24 under this rule? Because we will need a way to distinguish those (in Factors & Multiples), but still regard 2 * 2 * 2 * 3 to be the same as 3 * 2 * 2 * 2. I think the Android version of MatchesUpToTrivialManipulations achieves this, but what you just mentioned worries me a bit and may indicate that some further investigation is needed here.

Reply:
> would you then consider 2 * 2 * 2 * 3 to be the same as 24 under this rule?
No it won't. This is a purely numeric term and the numeric expression's functions wil handle this. So 2 * 2 * 2 * 3 will match up to trivial manipulations with answers like 2 * 2 * 3 * 2 or 3 * 2 * 2 * 2, but not with answers like 24 or 2 * 4 * 3.

Reply:
But if we can do this sort of thing for numeric expressions, then what's stopping us from doing it for algebraic expressions as well (if we wanted to, pending Ben's confirmation)?

Reply:
> @seanlip The `MatchesUpToTrivialManipulations` rule is the same as the initially implemented `MatchesExactlyWith` rule. And as per that rule's definition, we allow numeric simplification for algebraic expressions and equations. If we want to restrict that, we won't be able to use nerdamer since it automatically simplifies numeric parts.

I don't think this designation is correct. Per https://docs.google.com/document/d/1x2vcSjocJUXkwwlce5Gjjq_Z83ykVIn2Fp0BcnrOrTg/edit#heading=h.g0bd549g079d 'trivial manipulations' is defined as any reordering of terms based on associativity and commutativity such that the expression is still equivalent. If you look at the failure cases both in those tables and in [the test matrix](https://docs.google.com/spreadsheets/d/1u1fQdah2WsmdYKWKGmuXy5TPT7Ot-b8A7O9iZF-j5XE/edit#gid=0&range=I124) this specifically case shouldn't be allowed. Any sort of actual term combining only happens in the equivalence rule (since that cares more about numerical equivalence than verifying terms unlike the two 'matches' classifiers).

Reply:
Thanks @BenHenning. @iamprayush -- any thoughts on the above?

Reply:
@seanlip Fixed it so that algebraic expression input follows the same pattern.

Reply:
But then this particular test should evaluate to false, right?

Reply:
Yes, I've implemented the workaround that we'd discussed and this case evaluates to false now.